### ISSUE TITLE
Vivint CT200 Thermostat Driver

### ISSUE THREAD

**Issue by austinarchibald** [Botched1/Hubitat#15]
I was able to get it working, mostly. The only thing that is not showing is current temperature. It shows battery, correct set points, humidity, I can change the set points and modes from either Hubitat or from the device, and both will update... But current state shows no current temperature. So when the device shows in HomeKit, it just has a default temperature of 50°. I guess that's purely visual, since I know it's working as expected behind the scenes with my temperature set points, but still would like to figure this out. When I first set "Type" to User: Vivint CT200 Thermostat, no battery level would show. When I hit "refresh" command, then battery showed.

Any ideas?

**Comment by Botched1** [comment]
Unfortunately, no idea. 

As long as the thermostat has been properly excluded from a vivint panel it should work. If it was paired to a vivint panel, and not excluded, then it will not work until that is done due to Vivint using the zwave Anti-Theft Command Class. But I don't think that is your problem, as when that is the case you can't really control it at all over zwave - and it sounds like you can.

Past that, I don't know. I don't even have one of these thermostats anymore to test it against. Sorry!

**Comment by mitchjs** [comment]
@austinarchibald, i use this thermostat
I had to make changes to the driver code to make it compatible with the built in thermostat app
but i never had any issue getting current temp
keep in mind i dont think it gets current values on its own...
(i dont really remember) but the built in thermostat scheduler app, i have set to refresh every 5 mins
i could send you my modified code...
i made a gist: https://gist.github.com/mitchjs/1e4810caf08d4169a4eeda5c219e15f8

**Comment by mitchjs** [comment]
after setting the device to this driver, do a "configure" and then "refresh"

**Comment by austinarchibald** [comment]
@mitchjs thanks! Weird, I just went to update the code with yours, and before even doing so, temperature is now showing, so I guess it took a couple days for it to fix itself. I updated with your code, thanks for sharing it. All should be good for now. 

I have a Honeywell T6 that I am testing and comparing to the Vivint VS-ELEMO2-001 (I think it's the V2 or updated model for the CT200). I set a schedule (day/night) for the Honeywell, but for the Vivint, I don't see a way to set a schedule on the device itself - I guess I should just do this via Hubitat/Homekit?

**Comment by mitchjs** [comment]
yes i believe the device has no schedule, which is not a bad thing... when you got home automation
just use the built in thermostat scheduler app within Hubitat
i chose the vivint based on looks, its one of the few i can tolerate the look
the  honeywell t6 isnt horrible looking

im having zero issue with the CT200 (after i tweaked it)
the scheduling in the Hubitat is super flexible
![image](https://github.com/Botched1/Hubitat/assets/747871/1c2f79d7-9a2e-489b-922e-dcc5e0e68b8d)

**Comment by mitchjs** [comment]
im in Florida so its never used for HEATING :)

**Comment by austinarchibald** [comment]
This is perfect, thank you for the help and picture! I will do this and see how it goes. Yes, the Vivint one is probably the best looking Z-Wave device on the market. Just want to make sure it works well and that the drivers + hubitat + HomeKit plugin combo is reliable. So far so good once the current Temperature state was fixed, then add in your updated code with the scheduler, we should be good to go.
